### PR TITLE
[SYCL] Add support of 1 byte type to load and store subgroup operations

### PR DESCRIPTION
**Pull request by dkhaldi**
Signed-off-by: Dounia <dounia.khaldi@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/intel/sub_group.hpp [hunk 1]
```diff
@@ -280,18 +280,25 @@ struct sub_group {
...
     }
-    uint16_t t = __spirv_SubgroupBlockReadINTEL<uint16_t>(
-        (const __attribute__((ocl_global)) uint16_t *)src.get());
-    return *((T *)(&t));
+    else if (sizeof(T) == sizeof(uint16_t)) {
+      uint16_t t = __spirv_SubgroupBlockReadINTEL<uint16_t>(
+          (const __attribute__((ocl_global)) uint16_t *)src.get());
+      return *((T *)(&t));
+    }
+    else {uint8_t t = __spirv_SubgroupBlockReadINTEL<uint8_t>(
```
Comment (AlexeySachkov) [inline sycl/include/CL/sycl/intel/sub_group.hpp]: Formatting is broken here

File: sycl/test/sub_group/load_store.cpp [hunk 2]
```diff
@@ -145,6 +145,8 @@ int main() {
     return 0;
   }
   if (Queue.get_device().has_extension("cl_intel_subgroups")) {
+    typedef bool aligned_bool __attribute__((aligned(16)));
```
Comment (romanovvlad) [inline sycl/test/sub_group/load_store.cpp]: Why alignment here is 16?
Why do you expect sizeof(bool) to be 1 byte? especially on the device side?

**Comment by romanovvlad** [comment]
@dkhaldi ping.

File: sycl/test/sub_group/load_store.cpp (hunk 2 above)
Comment (dkhaldi) [reply sycl/test/sub_group/load_store.cpp]: @romanovvlad 
AFAIK, size of bool is 8 bits. Are you suggesting I use size_of(bool) in the code? The code currently implement 16 and 32. I am adding 8. Moreover, this also depends on the available spirv builtins where we only have 16, 32, and 8. So independently from bool, if spirv implement these 3, I think we should add the 3 to subgroup.hpp.

As for alignment 16, I just did the same as the other types in the test.

Comment (romanovvlad) [reply sycl/test/sub_group/load_store.cpp]: Sizeof bool: https://stackoverflow.com/questions/4897844/is-sizeofbool-defined-in-the-c-language-standard.
I think it's OK to provide implementation of the builtins for 8 bits types. It seems the SYCL specification requires that sizeof(bool) on host == sizeof(bool) on device, but I can't find it says sizeof(bool) == 1.
Why not use char instead which is guaranteed to 1 byte by the spec?

File: sycl/include/CL/sycl/intel/sub_group.hpp [hunk 3]
```diff
@@ -280,18 +280,26 @@ struct sub_group {
...
   load(const multi_ptr<T, Space> src) const {
     if (sizeof(T) == sizeof(uint32_t)) {
       uint32_t t = __spirv_SubgroupBlockReadINTEL<uint32_t>(
           (const __attribute__((ocl_global)) uint32_t *)src.get());
       return *((T *)(&t));
     }
-    uint16_t t = __spirv_SubgroupBlockReadINTEL<uint16_t>(
-        (const __attribute__((ocl_global)) uint16_t *)src.get());
-    return *((T *)(&t));
+    else if (sizeof(T) == sizeof(uint16_t)) {
```
Comment (bader) [inline sycl/include/CL/sycl/intel/sub_group.hpp]: Please, do not use `else` after `return`.
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

Also, apply clang-format to fix formatting.

**Pull request merged**